src/server/qtquick/wsurfaceitem.cpp

```diff
@@ -427,6 +427,19 @@ WSurface *WSurfaceItem::surface() const
 void WSurfaceItem::setSurface(WSurface *surface)
 {
     Q_D(WSurfaceItem);
+
+    if (!surface) {
+        // call setSurface(nullptr) to clear resources
+        // don't check `d->surface == surface` because original surface may destroyed
+        auto tmpFlags = d->surfaceFlags;
+        d->surfaceFlags = DontCacheLastBuffer; // clear up contentItem
```

Comment:
Why? If needs `DontCacheLastBuffer`, you should using it in QML.

Reply:
Set in QML works for dragicon, but shouldn‘t  we clear content cache when setSurface？